src/collections/Table/Table.js

```diff
@@ -1,23 +1,13 @@
 import _ from 'lodash';
 import React, {Children, Component, PropTypes} from 'react';
 import classNames from 'classnames';
-import TableColumn from './TableColumn';
+import {childrenOfTypesOnly} from '../../utils/customPropTypes';
 import META from 'src/utils/Meta';
 
 export default class Table extends Component {
   static propTypes = {
     basic: PropTypes.bool,
-    children: (props, propName, componentName) => {
-      let isValid = true;
-      Children.forEach(props.children, (child) => {
-        if (child.type !== TableColumn) {
-          isValid = false;
-        }
-      });
-      if (!isValid) {
-        return new Error('`Table` must only have `TableColumn` children.');
-      }
-    },
+    children: childrenOfTypesOnly(['TableColumn']),
```

Comment:
What's strange is the issue is closed but things seem not to work as expected. I know FB is inching away from `createClass`, yet most of the docs suggest many of the prop validators don't function otherwise (i.e. under ES6) classes. Though in my work on React Native I hadn't done much more complex than, for example: